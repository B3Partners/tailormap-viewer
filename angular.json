{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "cli": {
    "schematicCollections": [
      "@tailormap-viewer/schematics"
    ],
    "analytics": false
  },
  "projects": {
    "app": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:application": {
          "strict": true
        }
      },
      "root": "projects/app",
      "sourceRoot": "projects/app/src",
      "prefix": "app",
      "i18n": {
        "sourceLocale": "en-US",
        "locales": {
          "nl": {
            "translation": "locale/messages.nl.xlf"
          }
        }
      },
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/app",
            "index": "projects/app/src/index.html",
            "main": "projects/app/src/main.ts",
            "polyfills": "projects/app/src/polyfills.ts",
            "assets": [
              {
                "glob": "**/*",
                "input": "projects/app/src/assets",
                "output": "."
              },
              { "glob": "version.json", "input": "generated", "output": "." },
              {
                "glob": "**/*",
                "input": "projects/core/assets/icons",
                "output": "icons"
              }
            ],
            "styles": [
              "projects/core/assets/custom-theme.scss",
              "projects/core/assets/tailormap-styles.css"
            ],
            "scripts": [],
            "allowedCommonJsDependencies": [
              "xml-utils",
              "pbf",
              "html2canvas",
              "core-js",
              "raf",
              "rgbcolor",
              "lru-cache",
              "web-worker",
              "dompurify",
              "rbush",
              "cssesc",
              "font-family-papandreou"
            ],
            "i18nMissingTranslation": "warning"
          },
          "configurations": {
            "nl": {
              "localize": ["nl"]
            },
            "production": {
              "tsConfig": "projects/app/tsconfig.prod.json",
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "2.5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "4kb",
                  "maximumError": "6kb"
                }
              ],
              "fileReplacements": [
                {
                  "replace": "projects/app/src/environments/environment.ts",
                  "with": "projects/app/src/environments/environment.prod.ts"
                }
              ],
              "outputHashing": "all",
              "sourceMap": {
                "styles": false,
                "scripts": true,
                "vendor": true,
                "hidden": false
              }
            },
            "development": {
              "tsConfig": "projects/app/tsconfig.app.json",
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "browserTarget": "app:build:production"
            },
            "development": {
              "browserTarget": "app:build:development"
            },
            "nl": {
              "browserTarget": "app:build:development,nl"
            }
          },
          "options": {
            "proxyConfig": "proxy.config.js"
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "app:build"
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "projects/app/**/*.ts",
              "projects/app/**/*.html"
            ]
          }
        }
      }
    },
    "core": {
      "projectType": "library",
      "root": "projects/core",
      "sourceRoot": "projects/core/src",
      "prefix": "tm",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:ng-packagr",
          "options": {
            "project": "projects/core/ng-package.json",
            "tsConfig": "projects/core/tsconfig.lib.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "projects/core/tsconfig.lib.prod.json"
            },
            "development": {
              "tsConfig": "projects/core/tsconfig.lib.json"
            }
          },
          "defaultConfiguration": "production"
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "projects/core/**/*.ts",
              "projects/core/**/*.html"
            ]
          }
        }
      }
    },
    "api": {
      "projectType": "library",
      "root": "projects/api",
      "sourceRoot": "projects/api/src",
      "prefix": "tm",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:ng-packagr",
          "options": {
            "project": "projects/api/ng-package.json",
            "tsConfig": "projects/api/tsconfig.lib.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "projects/api/tsconfig.lib.prod.json"
            },
            "development": {
              "tsConfig": "projects/api/tsconfig.lib.json"
            }
          },
          "defaultConfiguration": "production"
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "projects/api/**/*.ts",
              "projects/api/**/*.html"
            ]
          }
        }
      }
    },
    "shared": {
      "projectType": "library",
      "root": "projects/shared",
      "sourceRoot": "projects/shared/src",
      "prefix": "tm",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:ng-packagr",
          "options": {
            "project": "projects/shared/ng-package.json",
            "tsConfig": "projects/shared/tsconfig.lib.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "projects/shared/tsconfig.lib.prod.json"
            },
            "development": {
              "tsConfig": "projects/shared/tsconfig.lib.json"
            }
          },
          "defaultConfiguration": "production"
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "projects/shared/**/*.ts",
              "projects/shared/**/*.html"
            ]
          }
        }
      }
    },
    "map": {
      "projectType": "library",
      "root": "projects/map",
      "sourceRoot": "projects/map/src",
      "prefix": "lib",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:ng-packagr",
          "options": {
            "project": "projects/map/ng-package.json",
            "tsConfig": "projects/map/tsconfig.lib.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "projects/map/tsconfig.lib.prod.json"
            },
            "development": {
              "tsConfig": "projects/map/tsconfig.lib.json"
            }
          },
          "defaultConfiguration": "production"
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "projects/map/**/*.ts",
              "projects/map/**/*.html"
            ]
          }
        }
      }
    },
    "admin-app": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:application": {
          "strict": true
        }
      },
      "root": "projects/admin-app",
      "sourceRoot": "projects/admin-app/src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/app/admin",
            "index": "projects/admin-app/src/index.html",
            "main": "projects/admin-app/src/main.ts",
            "polyfills": "projects/admin-app/src/polyfills.ts",
            "tsConfig": "projects/admin-app/tsconfig.app.json",
            "assets": [
              {
                "glob": "**/*",
                "input": "projects/admin-app/src/assets",
                "output": "."
              },
              {
                "glob": "**/*",
                "input": "projects/core/assets/icons",
                "output": "icons"
              },
              {
                "glob": "**/*",
                "input": "projects/admin-core/assets/icons",
                "output": "icons"
              }
            ],
            "styles": [
              "projects/core/assets/custom-theme.scss",
              "projects/core/assets/tailormap-styles.css"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "1.5mb",
                  "maximumError": "2mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "fileReplacements": [
                {
                  "replace": "projects/admin-app/src/environments/environment.ts",
                  "with": "projects/admin-app/src/environments/environment.prod.ts"
                }
              ],
              "outputHashing": "all",
              "sourceMap": {
                "styles": false,
                "scripts": true,
                "vendor": true,
                "hidden": false
              }
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "browserTarget": "admin-app:build:production"
            },
            "development": {
              "browserTarget": "admin-app:build:development"
            }
          },
          "options": {
            "proxyConfig": "proxy.config.js"
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "admin-app:build"
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "projects/admin-app/**/*.ts",
              "projects/admin-app/**/*.html"
            ]
          }
        }
      }
    },
    "admin-core": {
      "projectType": "library",
      "root": "projects/admin-core",
      "sourceRoot": "projects/admin-core/src",
      "prefix": "tm-admin",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:ng-packagr",
          "options": {
            "project": "projects/admin-core/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "projects/admin-core/tsconfig.lib.prod.json"
            },
            "development": {
              "tsConfig": "projects/admin-core/tsconfig.lib.json"
            }
          },
          "defaultConfiguration": "production"
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "projects/admin-core/**/*.ts",
              "projects/admin-core/**/*.html"
            ]
          }
        }
      }
    },
    "admin-api": {
      "projectType": "library",
      "root": "projects/admin-api",
      "sourceRoot": "projects/admin-api/src",
      "prefix": "tm",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:ng-packagr",
          "options": {
            "project": "projects/admin-api/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "projects/admin-api/tsconfig.lib.prod.json"
            },
            "development": {
              "tsConfig": "projects/admin-api/tsconfig.lib.json"
            }
          },
          "defaultConfiguration": "production"
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "projects/admin-core/**/*.ts",
              "projects/admin-core/**/*.html"
            ]
          }
        }
      }
    }
  }
}
