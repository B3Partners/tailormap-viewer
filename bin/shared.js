const { spawn, execSync } = require('child_process');
const path = require('path');
const inquirer = require('inquirer');

const checkCleanGitRepo = () => {
  const gitStatus = execSync('git status --short').toString();
  const gitDirty = gitStatus !== '';
  if (gitDirty) {
    console.error('Git repository is dirty, please commit first before making a new release');
    process.exit();
  }
};

const requestProject = (message, callback) => {
  inquirer.prompt([{
    type: 'list',
    name: 'project',
    message: message,
    choices: ['api', 'core', 'map', 'shared'],
  }])
    .then(answers => {
      const project = answers.project;
      if (!project) {
        console.error('Please select a project');
        process.exit();
      }
      callback(project);
    });
};

const runCommand = (command, args, cwd) => {
  return new Promise((resolve, reject) => {
    const cmd = spawn(command, args, {
      stdio: 'inherit',
      env: process.env,
      cwd: cwd || path.resolve(__dirname, '../')
    });
    cmd.on('error', err => {
      console.error(err);
      reject();
    });
    cmd.on('close', (code) => {
      resolve();
    });
  });
};

exports.requestProject = requestProject;
exports.checkCleanGitRepo = checkCleanGitRepo;
exports.runCommand = runCommand;
