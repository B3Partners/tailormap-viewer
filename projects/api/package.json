{
  "name": "@tailormap-viewer/api",
  "version": "11.0.0",
  "peerDependencies": {
    "@angular/common": "^16.1.0",
    "@angular/core": "^16.1.0",
    "rxjs": "^7.8.0"
  },
  "dependencies": {
    "tslib": "^2.4.0"
  }
}