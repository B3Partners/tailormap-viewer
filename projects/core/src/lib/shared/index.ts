export * from './core-shared.module';
export * from './components/bottom-panel/bottom-panel.component';
