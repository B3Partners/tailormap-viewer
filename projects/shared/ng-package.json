{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/shared",
  "assets": [
    "../../locale/messages.shared.en.xlf",
    "../../locale/messages.shared.nl.xlf"
  ],
  "lib": {
    "entryFile": "src/public-api.ts"
  }
}
