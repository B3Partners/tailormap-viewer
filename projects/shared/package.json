{
  "name": "@tailormap-viewer/shared",
  "version": "11.0.9",
  "peerDependencies": {
    "@angular/animations": "^16.0.1",
    "@angular/cdk": "^16.0.1",
    "@angular/common": "^16.0.1",
    "@angular/core": "^16.0.1",
    "@angular/forms": "^16.0.1",
    "@angular/material": "^16.0.1",
    "@angular/platform-browser": "^16.0.1",
    "@ctrl/tinycolor": "^3.5.0",
    "rxjs": "^7.8.0",
    "@tinyhttp/content-disposition": "^2.1.0"
  },
  "dependencies": {
    "tslib": "^2.4.0"
  }
}