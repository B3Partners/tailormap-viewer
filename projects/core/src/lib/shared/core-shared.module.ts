import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BottomPanelComponent } from './components/bottom-panel/bottom-panel.component';
import { SharedModule } from '@tailormap-viewer/shared';

@NgModule({
  declarations: [
    BottomPanelComponent,
  ],
  imports: [
    CommonModule,
    SharedModule,
  ],
  exports: [
    BottomPanelComponent,
  ],
})
export class CoreSharedModule { }
