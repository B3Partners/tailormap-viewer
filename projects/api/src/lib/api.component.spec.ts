import {render, screen, fireEvent} from '@testing-library/angular';
import { ApiComponent } from './api.component';
import { SharedModule } from '@tailormap-viewer/shared';

describe('ApiComponent', () => {

  test('should create', async () => {
    const { container } = await render(ApiComponent, {
      imports: [ SharedModule ],
    });
    expect(container).toMatchSnapshot();
    expect(screen.getByText('api works!'));
  });

  test('should change the message after clicking', async () => {
    await render(ApiComponent, {
      imports: [ SharedModule ],
    });
    expect(screen.getByText('there'));
    fireEvent.click(screen.getByText('Update the message'));
    expect(screen.getByText('who are you?'));
  });

});
