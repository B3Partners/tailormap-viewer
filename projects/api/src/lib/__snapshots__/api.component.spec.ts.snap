// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiComponent should create 1`] = `
<div
  id="root0"
>
  <p>
     api works! 
    <span
      class="message"
    >
      there
    </span>
    <button>
      Update the message
    </button>
    <tm-shared>
      <button
        class="mat-focus-indicator mat-flat-button mat-button-base _mat-animation-noopable"
        mat-flat-button=""
      >
        <span
          class="mat-button-wrapper"
        >
          Test
        </span>
        <span
          class="mat-ripple mat-button-ripple"
          matripple=""
        />
        <span
          class="mat-button-focus-overlay"
        />
      </button>
      <span>
        Test2
      </span>
    </tm-shared>
  </p>
</div>
`;
